GoogleForCreators/web-stories-wp
# Media: Processing items missing after opening media dialog

## Context



## Summary

Add `onUpload` and `onDelete` callbacks to media upload component. This has a number of key benefits. This allow for media  added and removed from within the editor. For example, if an image is delete from the dialog, it is removed from the media library and on the canvas. 

I have also taken this opportunity to use this new `onDelete` to improve background audio and captions. If these files are deleted within the dialog, then background audio and caption are reset / removed. 

## Relevant Technical Choices



## To-do



## User-facing changes

https://user-images.githubusercontent.com/237508/179557163-02d60356-78a7-4eb7-ac6e-9083b1fcca83.mov

## Testing Instructions






This PR can be tested by following these steps:

1. Create a story. 
2. Open media dialog. 
3. Upload image. 
4. Insert into canvas. 
5. See new image into media library. 
---
1. Create a story. 
2. Open media dialog. 
3. Insert image into canvas. 
4. Open media dialog. 
5. Select image ( the one used before ). 
6. Delete image 
7. Close dialog. 
8. See image removed from canvas and in media library.
---
1. Create a story. 
2. Select page
3. Go to style tab. 
4. Upload audio file.
5. Select file. 
6. Click replace file. 
7. Delete image ( the one used before ). 
8. Close dialog. 
9. See background audio is reset. 

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #9963
